File: java/src/Polimorfismo/Esfera.java
package Polimorfismo;

public class Esfera extends Figuras{

    Esfera(String nome, double dim1) {
        this.nome = nome;
        this.dim1 = dim1;
    }
    public double area() {
        double area = 0;
        area = 4 * Math.PI * Math.pow(dim1, 2);
        return area;
    }

    public double volume() {
        double volume = 0;
        volume = 4 / 3 * Math.PI * Math.pow(dim1, 3);
        return volume;
    }
    void display(){
        System.out.println(nome + dim1);
    }

}

File: cpp/Polimorfismo/Esfera.cpp
#include <cmath>
#include <iostream>
#include "Figuras.h"

class Esfera : public Figuras {
public:
    Esfera(std::string nome, double dim1) : Figuras(nome, dim1) {}

    double area() override {
        return 4 * M_PI * std::pow(dim1, 2);
    }

    double volume() override {
        return (4.0 / 3) * M_PI * std::pow(dim1, 3);
    }

    void display() override {
        std::cout << nome << ": Raio = " << dim1 << std::endl;
    }

    void aumentar() override {
        dim1 *= aumento;
    }
};
